src/compiler/checker.ts

```diff
@@ -13595,7 +13595,30 @@ namespace ts {
         }
 
         function getRestTypeElementFlags(node: RestTypeNode | NamedTupleMember) {
-            return getArrayElementTypeNode(node.type) ? ElementFlags.Rest : ElementFlags.Variadic;
+            const typeNode = node.type;
+
+            switch (typeNode.kind) {
+                case SyntaxKind.TypeReference:
+                    const typeName = (typeNode as TypeReferenceNode).typeName;
+                    if (isIdentifier(typeName)) {
+                        const s = resolveName(typeName, typeName.escapedText, SymbolFlags.Type, undefined, typeName.escapedText, /*isUse*/ false);
+                        // @TODO: figure out why `isArrayLikeType(getTypeOfSymbol(s))` does not work

```

Comment:
`isArrayLikeType` only looks for a `length` property, first of all, so it too loose a check. Secondly, `getTypeOfSymbol` is for expressions, while here we're talking about a type reference - `getDeclaredTypeOfSymbol` is more appropriate, or, better yet, `getTypeFromTypeReference` on the type reference node directly. (Looking at the `.target` of the resulting type and seeing if it's the `globalArrayType` or `globalReadonlyArrayType` should be sufficient)